clojure-basics/src/learn_clojure/basics.clj

```diff
@@ -851,25 +851,31 @@ java.lang.Long/MAX_VALUE
 ;;
 ;; Some of the sequences produced by the core
 ;; library are lazy which means that the entire
-;; collection won't be create, but at first only
-;; the iterator like structure is created while
-;; the consumer is fetching the `next()` item it
-;; will be computed. This is a very important
-;; element of the language which allows to express
-;; easily infinite sequences without running out
+;; collection won't be created (*realised*) all at once.
+;; At first, an iterator like structure is created,
+;; with subsequent calls to `next()` causing chunks
+;; of items to be fetched/computed. This is a very important
+;; element of the language which allows the easy expression
+;; of infinite sequences without running out
 ;; of memory. For example the function `range`
 ;; returns a lazy sequence of natural numbers
 ;; between two given numbers. But when it is
 ;; called without arguments it returns a lazy
 ;; sequence of all natural numbers.  Yet it
 ;; doesn't run out of memory. What it really
-;; produce it is just a iterator that computes the
-;; next number when the `next()` is called.
-;;
+;; produces is just an iterator that computes the
+;; next chunk of numbers when `next()` is called.
+;; NOTE: As subsequent calls are made to `next()`,
```

Comment:
Given this is more of basic intro stuff. Don't know whether mentioning "head-holding" and chunking is 
bit too much